infoschema/tables.go

```diff
@@ -984,7 +984,7 @@ func dataForProcesslist(ctx sessionctx.Context) [][]types.Datum {
 	for _, pi := range pl {
 		// If you have the PROCESS privilege, you can see all threads.
 		// Otherwise, you can see only your own threads.
-		if !hasProcessPriv && pi.User != loginUser.Username {
+		if !hasProcessPriv && loginUser != nil && pi.User != loginUser.Username {
```

Comment:
I think that we should check the privileges in local TiDB instead of remote TiDB.

Reply:
No, the privileges should check in the remove, unless you add a `selection` to do privilege check in local.